sbom.cdx.json

```diff
@@ -0,0 +1,43 @@
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.6",
+  "version": 1,
+  "metadata": {
+    "authors": [
+      {
+        "name": "@VCS_SBOM_AUTHORS@"
+      }
+    ]
+  },
+  "components": [
+    {
+      "type": "library",
+      "bom-ref": "pkg:github/Mbed-TLS/mbedtls@@VCS_TAG@",
+      "cpe": "cpe:2.3:a:mbed:mbedtls:@VCS_TAG@:*:*:*:*:*:*:*",
+      "name": "mbedtls",
+      "version": "@VCS_VERSION@",
+      "description": "Implements cryptographic primitives, X.509 certificate manipulation and SSL/TLS and DTLS protocols",
+      "supplier": {
+        "name": "The Mbed TLS Contributors"
+      },
+      "licenses": [
+        {
+          "license": {
+            "id": "Apache-2.0"
+          }
+        },
+        {
+          "license": {
+            "id": "GPL-2.0-or-later"
+          }
+        }
+      ],
+      "externalReferences": [
+        {
+          "type": "vcs",
+          "url": "https://github.com/Mbed-TLS/mbedtls"
+        }
+      ]
+    }
```

Comment:
Should there be a dependency on `mbedtls-framework`? That repository contains build scripts (including source code generators). The dependency is implemented as a git submodule (`framework`).

Reply:
I think we need an extra sbom.cdx.json in there; I'm happy to do that too of course, but make get this one perfect and then the other one becomes a formality. The uswid CLI actually adds a dependency on subprojects automatically, but I can make it explicit too if you'd rather.

Reply:
> The uswid CLI actually adds a dependency on subprojects automatically

Good to know, thanks. So I think this works out:

* For Git consumers, they'll get the dependency automatically.
* For tarball consumers, they'll get a tarball with both `sbom.ddx.json` files. There won't be a dependency but the whole thing comes as a single tarball so it can't be dissociated.

Reply:
> For Git consumers, they'll get the dependency automatically.

Exactly.

> For tarball consumers, they'll get a tarball with both sbom.ddx.json files

Aha. So if we do want to ship them in the tarball we should add the dep section manually then; uswid will only auto-add the dep if it's not already present. Which way around makes sense, _mbedtls requires mbedtls-framework_ I assume? Or do they depend on each other the "other way round" too?

Reply:
`mbedtls-framework` (the `framework` submodule) does not provide a useful product on its own. It's used as part of the build of `mbedtls`. The framework is mostly test code, but it also contains a few build scripts, so a tainted framework would taint the library build.